### PR TITLE
benchmarks: add store libary to comparison.

### PR DESCRIPTION
**Pull request by qnikst**
Store library was added to comparison.
Currently benchmarks for store are using generic instances,
so possibly using manual instances may improve results.

All benchmarks for store uses strict bytestrings,
this may be not fair but doesn't change results.

deserialiseNul benchmark is missing at the moment.

### CODE REVIEW

**Comment by dcoutts** [comment]
Thanks!

File: bench/versus/Micro.hs [hunk 1]
```diff
@@ -60,6 +61,7 @@ benchmarks =
       , bench "aeson generic" (nf perfDecodeAesonGeneric tstdataA)
       , bench "aeson TH"      (nf perfDecodeAesonTH      tstdataA)
       , bench "read/show"     (nf perfDecodeReadShow     tstdataS)
+      , bench "read/show"     (nf perfDecodeReadShow     tstdataS)
```
Comment (dcoutts) [inline bench/versus/Micro.hs]: Is this right? looks like a dupe

File: bench/versus/Micro.hs [hunk 2]
```diff
@@ -80,6 +82,7 @@ benchmarks =
     tstdataS = Micro.ReadShow.serialise tstdata
 --  tstdataM = PkgMsgpack.serialise tstdata
 --  tstdataN = Micro.NewMsgpack.serialise tstdata
+    tstdataR = Micro.Store.serialise tsdata
```
Comment (dcoutts) [inline bench/versus/Micro.hs]: duplicating the var name?

File: bench/versus/Micro.hs [hunk 3]
```diff
@@ -98,6 +102,7 @@ benchmarks =
     perfDecodeAesonGeneric = Micro.PkgAesonGeneric.deserialise
     perfDecodeAesonTH      = Micro.PkgAesonTH.deserialise
     perfDecodeReadShow     = Micro.ReadShow.deserialise
+    perfDecodeReadShow     = Micro.PkgStore.deserialise
```
Comment (dcoutts) [inline bench/versus/Micro.hs]: dupe?

File: bench/versus/Micro.hs [hunk 4]
```diff
@@ -88,6 +91,7 @@ benchmarks =
     perfEncodeAesonGeneric = BS.length . Micro.PkgAesonGeneric.serialise
     perfEncodeAesonTH      = BS.length . Micro.PkgAesonTH.serialise
     perfEncodeReadShow     = BS.length . Micro.ReadShow.serialise
+    perfEncodeReadShow     = BS.length . Micro.PkgStore.serialise
```
Comment (dcoutts) [inline bench/versus/Micro.hs]: dupe?

File: bench/versus/Micro/PkgStore.hs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+import Micro.Types
+import Data.Store as Store
+import Data.ByteString.Lazy as BS
+
+serialise :: Tree -> BS.ByteString
+serialise pkgs = BS.fromStrict (Store.encode pkgs)
+
+deserialise :: BS.ByteString -> Tree
+deserialise = (\(Right x) -> x) . Store.decode . BS.toStrict
```
Comment (dcoutts) [inline bench/versus/Micro/PkgStore.hs]: Mm, we should think about how to do fair comparison between libs when one of them forces the use of strict bytestrings.

**Comment by dcoutts** [comment]
@thoughtpolice I don't understand how travis thinks this compiles. Is it compiling the benchmarks?

**Comment by qnikst** [comment]
@dcoutts it should compile because it doesn't load problematic files. I'll address your comments and double check everything. Sorry for sending PR with such problems.

**Comment by qnikst** [comment]
@dcoutts I have addressed all your comments. It appeared that they all were in a dead code, that have not been compiled when compiling benchmarks. I took responsibility for re-enabling that code in the same commit.

As for fair comparison, will it be ok  to use strict bytestrings to read file and convert that to lazy for libraries that want to have lazy input? Though I'm not sure about encoding.

**Comment by dcoutts** [comment]
> As for fair comparison, will it be ok to use strict bytestrings to read file and convert that to lazy for libraries that want to have lazy input?

Yes, that gives the same best conditions to each lib.

For output/encoding, it's ok to have each one use it's native output style, ie no need to convert lazy to strict.

**Comment by dcoutts** [comment]
@qnikst looks like the benchmark isn't building on travis.

```
bench/versus/Main.hs:10:18:
    Could not find module ‘Micro’
    Use -v to see a list of the files searched for.
```

**Comment by qnikst** [comment]
Right, that file was not added to sdist as was missing in other modules, I'll test building tests after sdist locally.

**Comment by qnikst** [comment]
@dcoutts 
1. I've moved to deserialisation to use  strict ByteStings.
2. Moved micro benchmarks versus store to `micro` benchmark
3. Remove unused files.

**Comment by thoughtpolice** [comment]
Sorry this took so long - merged!

**Pull request closed without merging**